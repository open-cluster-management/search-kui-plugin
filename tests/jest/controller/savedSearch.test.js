/*******************************************************************************
* Licensed Materials - Property of IBM
* (c) Copyright IBM Corporation 2019. All Rights Reserved.
*
* Note to U.S. Government Users Restricted Rights:
* Use, duplication or disclosure restricted by GSA ADP Schedule
* Contract with IBM Corp.
*******************************************************************************/
'use strict'
const search = require('../../../dist/src-web/controller/savedSearch')
const savedsearches = require('../../data/savedsearches')

const args = {
  argv: ['savedsearches'],
  command: 'savedsearches'
}

describe('Get query count', () => {
  const spy = jest.spyOn(search, 'getQueryCount')
  const _ = search.getQueryCount(savedsearches.data)

  expect(spy).toBeCalled()

  it('should get query count', () => {
    expect(search.getQueryCount(savedsearches.data)).toMatchSnapshot()
    expect(search.getQueryCount).toMatchSnapshot()
  })
})

describe('Saved Search command', () => {
  const spy = jest.spyOn(search, 'doSavedSearch')
  const _ = search.doSavedSearch(args)

  expect(spy).toBeCalled()

  it('should return saved searches', () => {
    expect(search.doSavedSearch(args)).toMatchSnapshot()
    expect(search.doSavedSearch).toMatchSnapshot()
  })
})

