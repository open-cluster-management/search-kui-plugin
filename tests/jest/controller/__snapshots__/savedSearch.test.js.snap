// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get query count should get query count 1`] = `Promise {}`;

exports[`Get query count should get query count 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "__typename": "userSearch",
          "description": "All pods",
          "id": "1573570796760",
          "name": "All pods",
          "searchText": "kind:pod",
        },
        Object {
          "__typename": "userSearch",
          "description": "All deployments",
          "id": "1573570796793",
          "name": "All deployments",
          "searchText": "kind:deployments",
        },
      ],
    ],
    Array [
      Array [
        Object {
          "__typename": "userSearch",
          "description": "All pods",
          "id": "1573570796760",
          "name": "All pods",
          "searchText": "kind:pod",
        },
        Object {
          "__typename": "userSearch",
          "description": "All deployments",
          "id": "1573570796793",
          "name": "All deployments",
          "searchText": "kind:deployments",
        },
      ],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`Saved Search command should return saved searches 1`] = `Promise {}`;

exports[`Saved Search command should return saved searches 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "argv": Array [
          "savedsearches",
        ],
        "command": "savedsearches",
      },
    ],
    Array [
      Object {
        "argv": Array [
          "savedsearches",
        ],
        "command": "savedsearches",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;
